### DISCUSSION TITLE
Custom colletion index?

### DISCUSSION THREAD

**Discussion by lesomnus** [devcontainers/spec#136] [Q&A] [answered]
I know that _devcontainer_ is now a specification independent of tools like _vscode_, but most maintainers seem to be related to _vscode_ too, so I'm asking based on my experience with _vscode_.
I tried to post a question to `microsoft/vscode-dev-containers`, but I apologize in advance for posting it here, as there is no question template and they do not accept feature requests.

Most of our team uses the _vscode_ _devcontainer_ feature, and there is a _devcotnainer_ template to use in our team. Is there a way to share a template other than adding it to the public index? Because our team is small, we don't think tempalte is suitable for being listed in the public index as it is not generally usable in many environments.

For example, I would like to see a list of devcontainer templates used in the team when running the "Reopen in Container" command in _vscode_.

According to the spec, it seems to get the collection list from the OCI registry. How about providing a guideline so that make the supporting tools (e.g. _vscode_) can set the OCI reference list of the collection to be referenced?

**Comment by joshspicer** [comment, marked as answer]
I think this is a great idea, and something I will bring up at our internal planning meeting.  This should be quite feasible to do technically.

**Comment by MLong-Sorenson** [comment]
Any updates? This would be a huge benefit for my development team as well. 

I'm curious to know why template functionality is limited to an OCI registry. We just want to implement a template using the existing process (https://github.com/devcontainers/spec/blob/main/proposals/devcontainer-templates.md#option-resolution), and then distribute a custom devcontainer-template.json within the same repo as a custom devcontainer.json.

**Comment by MarlonPassos-git** [comment]
@joshspicer some update?

**Reply by joshspicer** [reply]
Unfortunately this is not something the team has been able to prioritize
